Exception in function getLocalHostLANAddress() on FreeBSD

I run Atlassian Confluence 5.10 or 6.0 on JRE 1.8.0_102-b14 on FreeBSD 11, and it can't start because of exception. Can you please check, if function `getLocalHostLANAddress()` works correctly on FreeBSD? I've tried to search for solution, but can't find anything and have a little knowledge to debug myself.

```
Caused by: java.lang.IllegalStateException: Unable to get Inet address
        at com.github.kristofa.brave.Brave$Builder.<init>(Brave.java:59)
        at com.atlassian.confluence.util.zipkin.impl.BraveFactoryBean.createInstance(BraveFactoryBean.java:32)
        at org.springframework.beans.factory.config.AbstractFactoryBean.afterPropertiesSet(AbstractFactoryBean.java:134)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1637)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1574)
        ... 107 more
Caused by: java.net.UnknownHostException: Failed to determine LAN address: java.net.SocketException: Cannot assign requested address
        at com.github.kristofa.brave.InetAddressUtilities.getLocalHostLANAddress(InetAddressUtilities.java:80)
        at com.github.kristofa.brave.Brave$Builder.<init>(Brave.java:56)
        ... 111 more
Caused by: java.net.SocketException: Cannot assign requested address
        at java.net.NetworkInterface.getAll(Native Method)
        at java.net.NetworkInterface.getNetworkInterfaces(NetworkInterface.java:343)
        at com.github.kristofa.brave.InetAddressUtilities.getLocalHostLANAddress(InetAddressUtilities.java:43)
        ... 112 more
```

--- a/brave-core/src/main/java/com/github/kristofa/brave/Brave.java
+++ b/brave-core/src/main/java/com/github/kristofa/brave/Brave.java
@@ -10,6 +10,8 @@ import com.twitter.zipkin.gen.Endpoint;
 import com.twitter.zipkin.gen.Span;
 import java.net.UnknownHostException;
 import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 import zipkin.Constants;
 import zipkin.reporter.AsyncReporter;
 import zipkin.reporter.Reporter;
@@ -20,7 +22,6 @@ import static com.github.kristofa.brave.InetAddressUtilities.toInt;
 import static zipkin.internal.Util.checkNotNull;
 
 public class Brave {
-
     private final Clock clock;
     private final ServerTracer serverTracer;
     private final ClientTracer clientTracer;
@@ -44,9 +45,10 @@ public class Brave {
      * </ul>
      */
     public static class Builder {
+        final Logger logger = Logger.getLogger(Brave.class.getName());
         final SpanFactory.Default.Builder spanFactoryBuilder = SpanFactory.Default.builder();
         private final ServerClientAndLocalSpanState state;
-        private final Endpoint localEndpoint;
+        final Endpoint localEndpoint;
         private boolean allowNestedLocalSpans = false;
         private Clock clock;
         private Recorder recorder;
@@ -76,13 +78,14 @@ public class Brave {
          * @param serviceName Name of the local service being traced. Should be lowercase and not <code>null</code> or empty.
          */
         public Builder(String serviceName) {
+            int ipv4 = 127 << 24 | 1;
             try {
-                int ip = toInt(getLocalHostLANAddress());
-                state = new ThreadLocalServerClientAndLocalSpanState(ip, 0, serviceName);
-                localEndpoint = state.endpoint();
+                ipv4 = toInt(getLocalHostLANAddress());
             } catch (UnknownHostException e) {
-                throw new IllegalStateException("Unable to get Inet address", e);
+                logger.log(Level.WARNING, "Unable to get Inet address", e);
             }
+            state = new ThreadLocalServerClientAndLocalSpanState(ipv4, 0, serviceName);
+            localEndpoint = state.endpoint();
         }
 
         /**

--- a/brave-core/src/test/java/com/github/kristofa/brave/BraveTest.java
+++ b/brave-core/src/test/java/com/github/kristofa/brave/BraveTest.java
@@ -1,14 +1,27 @@
 package com.github.kristofa.brave;
 
 import com.github.kristofa.brave.AnnotationSubmitter.Clock;
+import com.twitter.zipkin.gen.Endpoint;
 import com.twitter.zipkin.gen.Span;
+import java.net.NetworkInterface;
+import java.net.SocketException;
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertSame;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
 
+@RunWith(PowerMockRunner.class)
+// Added to declutter console: tells power mock not to mess with implicit classes we aren't testing
+@PowerMockIgnore({"org.apache.logging.*", "javax.script.*"})
+@PrepareForTest({InetAddressUtilities.class, NetworkInterface.class})
 public class BraveTest {
   Brave brave = newBrave();
 
@@ -125,5 +138,13 @@ public class BraveTest {
   @Test
   public void testGetServerSpanThreadBinder() {
     assertNotNull(brave.serverSpanThreadBinder());
+  }
+
+  @Test public void implicitLocalEndpoint_exceptionReadingNics() throws Exception {
+    mockStatic(NetworkInterface.class);
+    when(NetworkInterface.getNetworkInterfaces()).thenThrow(SocketException.class);
+
+    assertThat(new Brave.Builder().localEndpoint)
+        .isEqualTo(Endpoint.create("unknown", 127 << 24 | 1));
   }
 }

